overlay.d/05rhcos/usr/lib/dracut/modules.d/41rhcos-early-reboot/coreos-kargs-reboot.service

```diff
@@ -0,0 +1,24 @@
+[Unit]
+Description=CoreOS Early Reboot
+ConditionPathExists=/etc/initrd-release
+DefaultDependencies=false
+Before=ignition-complete.target
+ 
+# This runs after ignition-kargs & before ignition-disks so that it can optionally reboot
+# if kargs were modified via Ignition. This is done in a two-stage fashion so that other
+# mechanisms which may want to reboot (e.x. FIPS) can also hook in here and only reboot
+# once from the initrd.
+After=ignition-kargs.service
+Before=ignition-disks.service
+
+# On RHCOS we also need to run after FIPS and reboot in the case that
+# either Ignition kargs or FIPS has performed modifications.
+After=rhcos-fips.service
```

Comment:
Totally fine as is, though in the future we could add our own little synthetic `coreos-ignition-extra.target` or something?  OTOH, hopefully it's unlikely that we discover a 3rd reason to reboot.